apps/dashboard/src/components/primitives/field-editor/variable-popover/variable-popover.tsx

```diff
@@ -0,0 +1,257 @@
+  const [name, setName] = useState(parsedName);
+  const [defaultVal, setDefaultVal] = useState(parsedDefaultValue);
+  const [showRawLiquid, setShowRawLiquid] = useState(false);
+  const [searchQuery, setSearchQuery] = useState('');
+
+  useEffect(() => {
+    const handleEscape = (e: KeyboardEvent) => {
+      if (e.key === 'Escape') {
+        onClose();
+      }
+    };
+
+    document.addEventListener('keydown', handleEscape);
+    return () => document.removeEventListener('keydown', handleEscape);
+  }, [onClose]);
+
+  const {
+    transformers,
+    dragOverIndex,
+    draggingItem,
+    setDragOverIndex,
+    setDraggingItem,
+    handleTransformerToggle,
+    moveTransformer,
+    handleParamChange,
+    getFilteredTransformers,
+  } = useTransformerManager({
+    initialTransformers: parsedTransformers,
+    onUpdate: (newTransformers) => {
+      onUpdate(formatLiquidVariable(name, defaultVal, newTransformers));
+    },
+  });
+
+  const handleNameChange = (newName: string) => {
+    setName(newName);
+    onUpdate(formatLiquidVariable(newName, defaultVal, transformers));
+  };
+
+  const handleDefaultValueChange = (newDefaultVal: string) => {
+    setDefaultVal(newDefaultVal);
+    onUpdate(formatLiquidVariable(name, newDefaultVal, transformers));
+  };
+
+  const handleRawLiquidChange = (value: string) => {
+    // Remove {{ and }} and trim
+    const content = value.replace(/^\{\{\s*|\s*\}\}$/g, '').trim();
+
+    // Split by pipe and trim each part
+    const parts = content.split('|').map((part) => part.trim());
+
+    // First part is the name
+    const newName = parts[0];
+    setName(newName);
+
+    // Process each part after the name
+    parts.slice(1).forEach((part) => {
+      if (part.startsWith('default:')) {
+        // Extract default value, handling quotes
+        const newDefaultVal = part
+          .replace('default:', '')
+          .trim()
+          .replace(/^["']|["']$/g, '');
+        setDefaultVal(newDefaultVal);
+      }
+    });
+  };
+
+  return (
+    <PopoverContent className="w-72 p-0">
+      <div>
+        <div className="bg-bg-weak">
+          <div className="flex flex-row items-center justify-between space-y-0 p-1.5">
+            <div className="flex items-center gap-1">
+              <span className="font-subheading-2x-small text-subheading-2xs text-text-soft">CONFIGURE VARIABLE</span>
+            </div>
+          </div>
+        </div>
+        <div className="grid gap-1.5">
+          <div className="grid gap-2 p-2">
+            <FormItem>
+              <FormControl>
+                <div className="grid gap-1">
+                  <label className="text-text-sub text-label-xs">Variable name</label>
+                  <InputField size="fit" className="min-h-0">
+                    <Code2 className="text-text-sub h-4 w-4" />
+                    <Input value={name} onChange={(e) => handleNameChange(e.target.value)} className="h-7 text-sm" />
```

Comment:
When there are modifiers and if the variable name gets removed all the modifiers appear in the editor.
I feel like we should only apply the changes when the popover is closed.

https://github.com/user-attachments/assets/33b2c496-ac7b-4c34-94a2-dfd25028513a